pylint/checkers/classes/class_checker.py

```diff
@@ -1560,15 +1560,31 @@ def visit_assignattr(self, node: nodes.AssignAttr) -> None:
     def _check_invalid_class_object(self, node: nodes.AssignAttr) -> None:
         if not node.attrname == "__class__":
             return
-        inferred = safe_infer(node.parent.value)
+        if isinstance(node.parent, nodes.Tuple):
+            class_index = -1
+            for i, elt in enumerate(node.parent.elts):
+                if hasattr(elt, "attrname") and elt.attrname == "__class__":
+                    class_index = i
+            if class_index == -1:
+                # This should not happen because we checked that the node name
+                # is '__class__' earlier, but let's not be too confident here
```

Comment:
Good to have this. Maybe assignment via nested tuples could have violated this assumption?